src/typeInference/annotatePathExpr.ts

```diff
@@ -9,11 +9,29 @@ import astHelper, { IAST } from '../parsing/astHelper';
  * @returns the inferred SequenceType
  */
 export function annotatePathExpr(ast: IAST): SequenceType {
-	const seqType = {
-		type: ValueType.ITEM,
-		mult: SequenceMultiplicity.ZERO_OR_MORE,
-	};
-
+	let seqType;
+	const children = astHelper.getChildren(ast, 'stepExpr');
+	const lastChild = children[children.length - 1];
+	if (children && lastChild) {
+		if (astHelper.followPath(lastChild, ['filterExpr'])) {
+			seqType = astHelper.getAttribute(
+				astHelper.followPath(lastChild, ['filterExpr', '*']),
+				'type'
+			) as SequenceType;
+		}
+		if (astHelper.followPath(lastChild, ['xpathAxis'])) {
+			seqType = {
+				type: ValueType.NODE,
+				mult: SequenceMultiplicity.ZERO_OR_MORE,
+			};
+		}
+	}
+	if (!seqType) {
+		seqType = {
+			type: ValueType.ITEM,
+			mult: SequenceMultiplicity.ZERO_OR_MORE,
+		};
```

Comment:
Is this the new treatment for the cases that we cannot infer the type? Item()* in another shell?

Reply:
This represents item()*